src/ai/backend/manager/models/utils.py

```diff
@@ -93,7 +93,6 @@ async def begin_session(self) -> AsyncIterator[SASession]:
             session = SASession(bind=conn)
             try:
                 yield session
```

Comment:
Instead of calling `session.commit()` every time in the async with context, how about adding keyword parameter to `begin_session()` which determines to do the commit or not? 
```suggestion
                yield session
                if do_commit:
                    await session.commit()
```